utils/instructions/Dual/index.ts

```diff
@@ -404,3 +432,227 @@ export async function getWithdrawInstruction({
+      }
+    }
+
+    const soName = `LSO-${form.optionExpirationUnixSeconds}`;
+    const [issueAuthority, _issueAuthorityBump] = await web3.PublicKey.findProgramAddress(
+      [
+        Buffer.from(utils.bytes.utf8.encode('LSO')),
+        toBeBytes(form.optionExpirationUnixSeconds),
+      ],
+      new PublicKey('DiPbvUUJkDhV9jFtQsDFnMEMRJyjW5iS6NMwoySiW8ki')
+    );
+
+    const configInstruction = await so.createConfigInstruction(
+      form.optionExpirationUnixSeconds,
+      form.optionExpirationUnixSeconds,
+      new BN(form.numTokens),
+      new BN(form.lotSize),
+      soName,
+      //use sol wallet as authority
+      form.payer.extensions.transferAddress!,
+      baseMint,
+      //use helper account as base account
+      helperTokenAccount.publicKey,
+      quoteMint,
+      form.quoteTreasury.pubkey,
+      form.payer.extensions.transferAddress!,
+      issueAuthority,
+    )
+
+    additionalSerializedInstructions.push(
+      serializeInstructionToBase64(configInstruction)
+    )
+
+    //after everything we close helper account
+    additionalSerializedInstructions.push(
+      serializeInstructionToBase64(
+        closeAccount({
+          source: helperTokenAccount.publicKey,
+          //sol wallet
+          destination: form.payer.extensions.transferAddress,
+          //owner governance or sol wallet same as baseTokenAccount
+          owner: form.baseTreasury.extensions.token?.account.owner,
+        })
+      )
+    )
+
+    return {
+      serializedInstruction,
+      isValid: true,
+      prerequisiteInstructions: prerequisiteInstructions,
+      prerequisiteInstructionsSigners: [helperTokenAccount],
+      governance: form.baseTreasury?.governance,
+      additionalSerializedInstructions,
+      chunkSplitByDefault: true,
+      chunkBy: 1,
+    }
+  }
+
+  const obj: UiInstruction = {
+    serializedInstruction,
+    isValid: false,
+    governance: form.baseTreasury?.governance,
+    additionalSerializedInstructions,
+    chunkSplitByDefault: true,
+    chunkBy: 1,
+  }
+  return obj
+}
+
+export async function getInitStrikeInstruction({
+  connection,
+  wallet,
+  form,
+  schema,
+  setFormErrors,
+}: InitStrikeArgs): Promise<UiInstruction> {
+  const isValid = await validateInstruction({ schema, form, setFormErrors })
+
+  const serializedInstruction = ''
+  const additionalSerializedInstructions: string[] = []
+  const prerequisiteInstructions: TransactionInstruction[] = []
+  if (
+    isValid &&
+    form.payer &&
+    form.soName &&
+    form.strikes &&
+    form.baseTreasury &&
+    wallet?.publicKey
+  ) {
+    const so = getStakingOptionsApi(connection)
+
+    const baseMint = form.baseTreasury.extensions.mint?.publicKey
+    if (!baseMint) {
+      return {
+        serializedInstruction,
+        isValid: false,
+        governance: form.baseTreasury?.governance,
+        additionalSerializedInstructions: [],
+      }
+    }
+
+    for (const strike of form.strikes.split(',')) {
```

Comment:
should this be within the check above for a valid SO?

Reply:
We could get thorough with the validations, but ultimately, this all will need to be redone anyways. The Staking Options instruction in realms has always been a hack job for the narrow use case